virttest/cartesian_config.py

```diff
@@ -538,10 +538,64 @@ def dump(self, indent, recurse=False):
 match_substitute = re.compile("\$\{(.+?)\}")
 
 
+def _drop_suffixes(d):
+    """
+    Merge suffixes for same var, or drop off unnecessary suffixes
+
+    This step returns a copy of a suffix flattened dictionary.
+    """
+    # dictionary `d_flat' is going to be the mutated copy of `d`
+    d_flat = d.copy()
+    for key in d:
+        if key in _reserved_keys:
+            continue
+
+        if not isinstance(key, tuple):
+            continue
+
+        try:
+            # This file was invoked through cmdline
+            options.skipdups
```

Comment:
Are these two lines needed? It seems the second line `skipdups = options.skipdups` would be enough to trigger the exception.

Reply:
Good point, I merely copied this code snippet out from the previous author but will add the performance and this improvement in a separate commit.